sqlx-postgres/src/migrate.rs

```diff
@@ -252,23 +252,27 @@ CREATE TABLE IF NOT EXISTS _sqlx_migrations (
         migration: &'m Migration,
     ) -> BoxFuture<'m, Result<Duration, MigrateError>> {
         Box::pin(async move {
-            // Use a single transaction for the actual migration script and the essential bookeeping so we never
-            // execute migrations twice. See https://github.com/launchbadge/sqlx/issues/1966.
-            let mut tx = self.begin().await?;
             let start = Instant::now();
 
-            let _ = tx.execute(&*migration.sql).await?;
+            // execute migration queries
+            if migration.no_tx {
+                let _ = self.execute(&*migration.sql).await?;
+            } else {
+                // Use a single transaction for the actual migration script and the essential bookeeping so we never
+                // execute migrations twice. See https://github.com/launchbadge/sqlx/issues/1966.
+                let mut tx = self.begin().await?;
+                let _ = tx.execute(&*migration.sql).await?;
+                tx.commit().await?;
+            }
+
+            let elapsed = start.elapsed();
 
             // language=SQL
             let _ = query(r#"DELETE FROM _sqlx_migrations WHERE version = $1"#)
                 .bind(migration.version)
-                .execute(&mut *tx)
+                .execute(self)
```

Comment:
This should not execute outside the transaction.

See what was done here: https://github.com/launchbadge/sqlx/blob/68da5aefea0e7c14635f484531887aca4e393660/sqlx-postgres/src/migrate.rs#L214-L225

and in `execute_migration()`: https://github.com/launchbadge/sqlx/blob/68da5aefea0e7c14635f484531887aca4e393660/sqlx-postgres/src/migrate.rs#L277-L300